var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -70,22 +74,25 @@ class Dealii(CMakePackage, CudaPackage):
         values=("default", "11", "14", "17"),
     )
     variant("doc", default=False, description="Compile with documentation")
-    variant("examples", default=True, description="Compile tutorial programs")
+    variant("examples", default=True, description="Install tutorial programs")
+    variant("examples_compile", default=True, description="Compile tutorial programs")
```

Comment:
Let's not have 2 variants for this. Can't we make `examples` "Compile and install tutorial programs"?

Reply:
Do I understand correctly that the "examples" are compiled as a test during the build, and what gets installed are the sources (I might have misunderstood what "examples" are for `dealii`)?

In general, variant enter in the hash of a spec, so in my opinion they should be reflected in the artifact that gets generated. If compiling the examples doesn't affect the install tree and is only a test, then it should be triggered in other ways (for instance with `self.tun_tests`) Does that make sense?

Reply:
It would be good to get the "new" release changes merged in quickly.

And resolve the "examples" issue in a separate PR. [i.e merge in current changes - or remove the "examples" variant changes from this PR - and merge in]

Reply:
If the tests pass, I would propose to just merge as-is and I will open another pull request for the examples.

Reply:
The test is clear - so will go ahead and merge this in.

Reply:
> Do I understand correctly that the "examples" are compiled as a test during the build, and what gets installed are the sources (I might have misunderstood what "examples" are for `dealii`)?
> 
> In general, variant enter in the hash of a spec, so in my opinion they should be reflected in the artifact that gets generated. If compiling the examples doesn't affect the install tree and is only a test, then it should be triggered in other ways (for instance with `self.tun_tests`) Does that make sense?

It's neither of this. If the examples are compiled, then they are also installed as binaries (along with the source files of the examples). But you also have a choice not to compile them, and to install just *the source files* of the examples (which is what users would tipically start from when developing their own applications).

Reply:
We can address this in a separate PR

Reply:
@luca-heltai The I would consider a non-boolean variant for them:
```python
variant("examples", values=("none", "sources", "binaries"), ...)
```